crates/editor/src/editor.rs

```diff
@@ -22590,6 +22655,12 @@ pub trait SemanticsProvider {
         cx: &mut App,
     ) -> Option<Task<anyhow::Result<Vec<InlayHint>>>>;
 
+    fn semantic_tokens(
+        &self,
+        buffer_handle: Entity<Buffer>,
```

Comment:
Looking at this signature, it seems that
range
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#semanticTokens_rangeRequest
requests are not supported?
How does this work (cpu-wise, latency-wise) on files like editor/src/editor_tests.rs when those are edited continuously?
I do not believe always querying a token request for the entire file is sustainable what if they do not support deltas?), as language servers can be slow and Zed might load them with such queries.

Reply:
I've removed all the range bits from this change.